### PR TITLE
Bugfix 6314/fix percent complete when switching GLs

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- in openTool() removed call to verifyGroupDataMatchesWithFs() before contextId was loaded - caused problems since changes where not saved.
- fixed bug in doBackgroundVerseEditsUpdates() to handle data returned by getCheckVerseEditsInGroupData() being grouped by groupId
- fixes for changing GLs in tN (since checks can be different in GLs):
  - fix to keep track of current GL.
  - fix to not copy group data if GL is same as last time.
  - fixed to preserve existing group data when checks are copied from tN helps
  - fix to not set value in reducer if check is not present in current GL.
  - fix to update check categories on GL change

#### Please include detailed Test instructions for your pull request:
- using this project: 
[en_ult_3jn_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3468385/en_ult_3jn_book.zip)
- should see 17% for tN with en GL
- launch tN to make sure it works.
- return to tools card and still should see the progress
- switch to tn GL to hi and should see higher percentage because fewer checks
- launch tN to make sure it works.
- return to tools card and still should see the progress

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6321)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (3792fdb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-3792fdb.deb) is ready for download.
[macOS build 2.0.0 (3792fdb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-3792fdb.dmg) is ready for download.
[Windows x64 build 2.0.0 (3792fdb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-3792fdb.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (3792fdb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-3792fdb.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (5cff2ac)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-5cff2ac.deb) is ready for download.
[macOS build 2.0.0 (5cff2ac)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-5cff2ac.dmg) is ready for download.
[Windows x64 build 2.0.0 (5cff2ac)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-5cff2ac.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (5cff2ac)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-5cff2ac.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (bf3394c)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-bf3394c.deb) is ready for download.
[macOS build 2.0.0 (bf3394c)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-bf3394c.dmg) is ready for download.
[Windows x64 build 2.0.0 (bf3394c)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-bf3394c.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (bf3394c)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-bf3394c.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (c875da6)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-c875da6.deb) is ready for download.
[macOS build 2.0.0 (c875da6)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-c875da6.dmg) is ready for download.
[Windows x64 build 2.0.0 (c875da6)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-c875da6.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (c875da6)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-c875da6.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (305b27b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-305b27b.deb) is ready for download.
[macOS build 2.0.0 (305b27b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-305b27b.dmg) is ready for download.
[Windows x64 build 2.0.0 (305b27b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-305b27b.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (305b27b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-305b27b.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (c22e5cd)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-c22e5cd.deb) is ready for download.
[macOS build 2.0.0 (c22e5cd)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-c22e5cd.dmg) is ready for download.
[Windows x64 build 2.0.0 (c22e5cd)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-c22e5cd.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (c22e5cd)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-c22e5cd.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (1a01e35)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-1a01e35.deb) is ready for download.
[macOS build 2.0.0 (1a01e35)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-1a01e35.dmg) is ready for download.
[Windows x64 build 2.0.0 (1a01e35)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-1a01e35.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (1a01e35)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-1a01e35.setup.exe) is ready for download.

File: src/js/helpers/ProjectAPI.js [hunk 1]
```diff
@@ -143,21 +146,78 @@ export default class ProjectAPI {
     return fs.readJsonSync(dataPath);
   }
 
+  /**
+   * returns true if contextIds match
+   * @param {Object} contextId1
+   * @param {Object} contextId2
+   * @return {boolean}
+   */
+  isMatchingContextId(contextId1, contextId2) {
```
Comment (mannycolon) [inline src/js/helpers/ProjectAPI.js]: This is only checking if the reference of a context id matches not the entire context id. This could be misleading

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (7b531cb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-7b531cb.deb) is ready for download.
[macOS build 2.0.0 (7b531cb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-7b531cb.dmg) is ready for download.
[Windows x64 build 2.0.0 (7b531cb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-7b531cb.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (7b531cb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-7b531cb.setup.exe) is ready for download.

File: src/js/helpers/ProjectAPI.js [hunk 2]
```diff
@@ -143,21 +146,78 @@ export default class ProjectAPI {
     return fs.readJsonSync(dataPath);
   }
 
+  /**
+   * returns true if contextIds match
```
Comment (mannycolon) [inline src/js/helpers/ProjectAPI.js]: Should the method description be updated as well?

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Review by PhotoNomad0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6321)* status: 0 of 15 files reviewed, 2 unresolved discussions (waiting on @mannycolon)

___
*[src/js/helpers/ProjectAPI.js, line 155 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6321#-LlMkhCbupmAWl7EVRsN-r7-155:-LlYO2sF4SnzqWV4KNU2:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/1a01e35f060d1bd6d196bc63d5234c46bc092d10/src/js/helpers/ProjectAPI.js#L155)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

This is only checking if the reference of a context id matches not the entire context id. This could be misleading 
</blockquote></details>

Done.
___
*[src/js/helpers/ProjectAPI.js, line 150 at r8](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6321#-LlMkhCbupmAWl7EVRsN-r8-150:-LlYO3Fc305QIbcl7lqa:b-896fix) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/7b531cbe519440a94f479a5660c79beabb175ca1/src/js/helpers/ProjectAPI.js#L150)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Should the method description be updated as well?
</blockquote></details>

Done.

**Comment by mannycolon** [comment]
@PhotoNomad0 I switched the tn card GL to hi and did not see higher percentage complete. It stayed the same. Used [en_ult_3jn_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3468385/en_ult_3jn_book.zip)

**Comment by mannycolon** [comment]
Same problems with this project
[hi_aaa_tit_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3469242/hi_aaa_tit_book.zip)

**Comment by PhotoNomad0** [comment]
> @PhotoNomad0 I switched the tn card GL to hi and did not see higher percentage complete. It stayed the same. Used [en_ult_3jn_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3468385/en_ult_3jn_book.zip)

If you select metronomy, it will go back to 17%

**Comment by PhotoNomad0** [comment]
> Same problems with this project
> [hi_aaa_tit_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3469242/hi_aaa_tit_book.zip)

Looks to me like it should have the same count

**Comment by dsm-git** [comment]
[Linux build 2.0.0 (4834188)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-linux-x64-2.0.0-4834188.deb) is ready for download.
[macOS build 2.0.0 (4834188)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-macos-x64-2.0.0-4834188.dmg) is ready for download.
[Windows x64 build 2.0.0 (4834188)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x64-2.0.0-4834188.setup.exe) is ready for download.
[Windows x32 build 2.0.0 (4834188)](https://tc-builds.door43.org/build/bugfix-mcleanb-6314/translationCore-win-x32-2.0.0-4834188.setup.exe) is ready for download.

**Comment by mannycolon** [comment]
> If you select metronomy, it will go back to 17%

@PhotoNomad0 That was not part of the test instructions

**Comment by PhotoNomad0** [comment]
> > If you select metronomy, it will go back to 17%
> 
> @PhotoNomad0 That was not part of the test instructions

because it was outside of the scope of the issue, should have left out the % progress numbers since they just caused confusion.  In your project you had this selection after selected hi and then went back to en: 

Screen Shot 2019-08-05 at 11 36 47 AM

**Pull request merged**